Silence redundant warnings about slow setup

amelchio opened the pull request:
## Description:

This warning can easily trigger on slow servers (like the Raspberry Pi) because it simply takes more than 10 seconds to get through all the setup.

As users can do nothing to avoid the issue, I propose to downgrade the severity.

**Related issue (if applicable):** [Forum post](https://community.home-assistant.io/t/hassio-io-10-second-warnings/23714)

## Checklist:

If the code does not interact with devices:
  - [X] Local tests with `tox` run successfully.
  - [X] Tests have been added to verify that the new code works.

amelchio committed:
Downgrade slow domain setup warning

balloob commented:
I don't like this. This is not a solution, this is just hiding the warning.

The problem is that on slow systems, we get too many false positives for slow loading components.

I assume that this is only for entity components and not for other components (group, automation etc).
Since entity components have their own warning around setup platform, maybe we shouldn't warn for slow entity component setup?

rofrantz commented:
Intel NUC, i7, 8G M2 memory, ssd is considered a slow system ?
Don't think so.
And its happening for a lot components and most of them perform just local api requests.

amelchio commented:
> I don't like this. This is not a solution, this is just hiding the warning.

That's fine, I also don't like it much :). Lowering the severity seemed less extreme but I actually wanted to remove the log line completely.

> The problem is that on slow systems, we get too many false positives for slow loading components.

While that is true, I believe the problem is the warning itself.

The warning measures wall clock time since the domain setup job was added. This time, however, includes all other tasks hogging the interpreter. With my Hass.io on Pi 3 setup, the CPU is at 100% for almost a minute during a restart. Tasks are ready to run but are not scheduled. All of them risk getting this warning attached even though they have not had much of a chance to run.

To be useful, the warning timer should exclude periods where the setup is ready to be scheduled but I don't think there is any way to measure that?

> I assume that this is only for entity components and not for other components (group, automation etc).

Group and automation vary significantly in runtime for me but apparently they never exceed 10s. I am not sure why (I do not know how asyncio schedules tasks) but I don't think they are _inherently_ immune to this issue.

So my proposal now is to remove the warning completely ... ?

Note that we also log the full wall clock time for each component setup, so not much is lost:
```
2017-12-30 19:25:28 INFO (MainThread) [homeassistant.setup] Setup of domain automation took 3.0 seconds.
```

Also note that I am not saying that one minute at 100% CPU is a great startup experience, just that this particular warning does not help to improve it.

balloob commented:
This warning helps us find slow components. So let's hide it for entity component as they have their own warning. Non entity components will still be timed and should always finish <10s. Like you said, non entity components like group/automation never have this problem (because they don't do I/O).

The way our setup works is that we schedule all setups at the same time on the event loop. Dependencies will cause certain setups to wait for others to finish (this is not included in the warning time).

The reason this warning exists is because if someone writes a breaking component that blocks forever, we will now print a warning. If it blocks, the log message of how long the setup has taken will never be printed. This has helped us track down issues in the past.

amelchio committed:
Revert "Downgrade slow domain setup warning"

This reverts commit 64472c006bb553c6bb75a024384361adad50d565.

amelchio committed:
Remove warning for entity components

amelchio committed:
Remove lint

amelchio committed:
Fix original test with extra call

amelchio commented:
For fun, I added a 500 KB template to an automation and got these logs:

```
$ egrep 'group| automation' home-assistant.log.slow-group
2018-01-02 04:26:14 INFO (MainThread) [homeassistant.setup] Setting up group
2018-01-02 04:27:21 WARNING (MainThread) [homeassistant.setup] Setup of group is taking over 10 seconds.
2018-01-02 04:27:21 INFO (MainThread) [homeassistant.setup] Setup of domain group took 67.7 seconds.
2018-01-02 04:27:22 INFO (MainThread) [homeassistant.setup] Setting up automation
2018-01-02 04:27:24 INFO (MainThread) [homeassistant.setup] Setup of domain automation took 2.2 seconds.
```

So it is not quite impossible for non-entity components to have this problem.

Anyway, I updated the PR as requested and this will surely improve the situation a lot.
